### DISCUSSION TITLE
GrblHAL with Ethernet connectivity

### DISCUSSION THREAD

**Discussion by cheechNC** [grblHAL/STM32F7xx#15] [General]
Wondering if this board NUCLEO-F756ZG has been used with ioSender with Ethernet. I tried the WebBuilder for STM32F7xx but the connection combo box does not have an ethernet option. Surely Im missing something.

A board like this looks most interesting as it has ethernet and is apparently compatible with existing break out boards/ shields designed for the arduino Uno.

**Comment by terjeio** [comment]
There are board maps for the [Protoneer](https://github.com/grblHAL/STM32F7xx/blob/main/Inc/protoneer_3.xx_map.h) and the [Uno](https://github.com/grblHAL/STM32F7xx/blob/main/Inc/uno_map.h) but I have not enabled them in the Web Builder yet.
Ethernet cannot be enabled since D11 (Z-limit) is used by the ethernet hardware. Another pin could be used instead - but which? D0 and D1 (UART RX/TX) are candidates, and A4 and A5 are too? And it has to be documented somehow...

Ideas?

**Comment by cheechNC** [comment]
thanks for quick reply. Sorry not up to speed with these things. Only been looking for alternatives a bit and stumbled upon this.
The hat that I have and Like is this https://www.yrcnc.com/

**Reply by terjeio** [reply]
I have now added Uno and Protoneer board maps to the Web Builder, and enabled ethernet for all builds.
Note that the F756 is a 3.3V device - you may have to modify the breakout board to avoid damaging it.

The yrcnc board is a better board since it has optocoupled inputs, there is also a buffer on the board unlike the Uno/Protoneer boards.
I have sent a message to yrcnc requesting information about the board, preferably the schematic.

Since the D11 pin is used by the Ethernet hardware I had to modify the board maps depending on the configuration, see the map files or use the $pins command to check the pin allocation.

**Reply by cheechNC** [reply]
Thanks @terjeio! I will get a F756 and try.  Yeah that little b.o.b. was a wowzer back when I "stumbled" upon it. 2018 I think it was. All I have is the available manual PDF. The creator was Leuben Hristov. I really hope they are around and reply. He may be on Linkedin.

I wonder if it would  be possible to have 4 axis off of the F756Nucleo? are there enough pins? a silly though is that I don't need the XYZ individual homming/limits input.. I normally loop my switches NC and use one input for XYZ homming; 
Another thing for me would be M3 and M4 outputs. Perhaps a way to use Mist or Cool as M4. But not super important as I can trick the post processor to do that.

Ethernet is a must;

**Reply by terjeio** [reply]
I have not yet heard back from yrcnc.

> He may be on Linkedin.

I am not any longer due to their sloppy handling of user data (email, password, ...).
If you can get hold of the schematic then that would be nice.

> I wonder if it would be possible to have 4 axis off of the F756Nucleo?

There are plenty of pins - most not broken out to the shield. The shield may support four axes already if it is based on the Uno or Protoneer map.

See the board maps for the [Uno](https://github.com/grblHAL/STM32F7xx/blob/main/Inc/uno_map.h) and/or the [Protoneer](https://github.com/grblHAL/STM32F7xx/blob/main/Inc/protoneer_3.xx_map.h) for pin assignments.
